pkg/sql/colexec/unnest/unnest.go

```diff
@@ -63,30 +60,109 @@ func Prepare(_ *process.Process, arg any) error {
 
 func Call(_ int, proc *process.Process, arg any) (bool, error) {
 	param := arg.(*Argument).Es
-	if param.isCol {
+	switch param.typ {
+	case "str":
```

Comment:
This has a very bad taste -- I wonder WHY should you care about param type?   It is an expression and get evaluated, shouldn't this be the same?

Yuanning, Long ran or Ming Song/Mo Chen, one of them must have a strong opinion on this.